bundles/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/internal/profiles/ModbusGainOffsetProfile.java

```diff
@@ -0,0 +1,248 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.modbus.internal.profiles;
+
+import java.math.BigDecimal;
+import java.util.Optional;
+
+import javax.measure.UnconvertibleException;
+import javax.measure.Unit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.profiles.ProfileCallback;
+import org.openhab.core.thing.profiles.ProfileContext;
+import org.openhab.core.thing.profiles.ProfileTypeUID;
+import org.openhab.core.thing.profiles.StateProfile;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.Type;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Profile for applying gain and offset to values.
+ *
+ * Output of the profile is
+ * - (incoming value + pre-gain-offset) * gain (update towards item)
+ * - (incoming value / gain) - pre-gain-offset (command from item)
+ *
+ * Gain can also specify unit of the result, converting otherwise bare numbers to ones with quantity.
+ *
+ *
+ * @author Sami Salonen - Initial contribution
+ */
+@NonNullByDefault
+public class ModbusGainOffsetProfile implements StateProfile {
+
+    private final Logger logger = LoggerFactory.getLogger(ModbusGainOffsetProfile.class);
+    private static final String PREGAIN_OFFSET_PARAM = "pre-gain-offset";
+    private static final String GAIN_PARAM = "gain";
+
+    private final ProfileCallback callback;
+    private final ProfileContext context;
+
+    private Optional<QuantityType<?>> pregainOffset;
+    private Optional<QuantityType<?>> gain;
+
+    public ModbusGainOffsetProfile(ProfileCallback callback, ProfileContext context) {
+        this.callback = callback;
+        this.context = context;
+        {
```

Comment:
What's the purpose of the braces?

Reply:
For me it improves readability (one block = one parameter) mainly. They also prevent accidental usage of variables related to the other parameter as the scope is closed.

I can change this if needed.

Reply:
I won't insist on changing it.